bodhi/server/scripts/approve_testing.py

```diff
@@ -110,18 +111,43 @@ def main(argv=sys.argv):
 
                 if update.autotime and update.days_in_testing >= update.stable_days:
                     print(f"Automatically marking {update.alias} as stable")
-                    update.set_request(db=db, action=UpdateRequest.stable, username="bodhi")
+                    # For now only rawhide update can be created using side tag
+                    # Do not ad the release.pending_stable_tag if the update
+                    # was created from a side tag.
+                    if update.release.composed_by_bodhi:
+                        update.set_request(db=db, action=UpdateRequest.stable, username="bodhi")
                     # For updates that are not included in composes run by bodhi itself,
                     # mark them as stable
-                    if not update.release.composed_by_bodhi:
+                    else:
+                        # Single and Multi build update
                         update.add_tag(update.release.stable_tag)
-                        update.remove_tag(update.release.pending_testing_tag)
-                        update.remove_tag(update.release.pending_stable_tag)
                         update.status = UpdateStatus.stable
-                        update.date_stable = datetime.datetime.utcnow()
                         update.request = None
-                        update.date_pushed = datetime.datetime.utcnow()
                         update.pushed = True
+                        update.date_stable = update.date_pushed = func.current_timestamp()
+                        update.comment(db, "This update has been submitted for stable by bodhi",
+                                       author=u'bodhi')
+
+                        # Multi build update
+                        if update.from_tag:
+                            # Merging the side tag should happen here
+                            pending_signing_tag = update.release.get_pending_signing_side_tag(
+                                update.from_tag)
+                            testing_tag = update.release.get_testing_side_tag(update.from_tag)
+
+                            update.remove_tag(pending_signing_tag)
+                            update.remove_tag(testing_tag)
+                            update.remove_tag(update.from_tag)
+
+                            koji = buildsys.get_session()
+                            koji.deleteTag(pending_signing_tag)
+                            koji.deleteTag(testing_tag)
+
+                            koji.removeSideTag(update.from_tag)
```

Comment:
Hm, should we document why we use deleteTag() in the first two and removeSideTag() in the third one?

Reply:
Isn't the name of the method enough ? `pending_signing_tag` and `testing_tag` are tags and `update.from_tag` is a side tag.

Reply:
@mohanboddu do you know if there is an actual difference in Koji between side tags and tags ? Could we just use deleteTag to delete a side tag ?

Reply:
Discussed on IRC the removeSideTag API also remove the build target on koji side. I have added a comment to make this explicit.